### PR TITLE
[XLA:CPU][oneDNN] Enable Matmul + BiasAdd + Elu / Tanh / Relu6 fusions for onednn

### PR DESCRIPTION
**Pull request by gaurides**
This PR enables MatMul  fusions with different activations viz. Elu, Tanh and Relu6.

### CODE REVIEW

File: xla/tests/onednn_matmul_test.cc [hunk 1]
```diff
@@ -923,6 +951,117 @@ TEST_F(MatmulTest, TestF32ConstantWeights) {
   )");
 }
 
+// MM + BiasAdd + Elu : FP32
+TEST_F(MatmulTest, BiasAddELUFusion_F32) {
+  const char* matmul_module_str = R"(
+  HloModule matmul.test.f32
+
```
Comment (jreiffers) [inline xla/tests/onednn_matmul_test.cc:958]: Can you please clean this up a bit?

- sort, remove unnecessary attributes
- fix names
- remove default layouts

**Review by jreiffers — APPROVED** [review]
Looks alright to me, but I have not much context about this code. Might want to get a review from someone who works on CPU as well.

File: xla/tests/onednn_matmul_test.cc (hunk 1 above)
Comment (gaurides) [reply xla/tests/onednn_matmul_test.cc:958]: done. @jreiffers can you please take a relook?

Comment (jreiffers) [reply xla/tests/onednn_matmul_test.cc:958]: Thank you very much!

**Review by jreiffers — APPROVED** [review]

File: xla/service/cpu/onednn_pattern_utils.h [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+#include "xla/hlo/ir/hlo_instruction.h"
+#include "xla/hlo/ir/hlo_instructions.h"
+#include "xla/service/cpu/onednn_util.h"
+#include "xla/service/pattern_matcher.h"
+
+namespace xla {
+namespace cpu {
+
+namespace {
```
Comment (jreiffers) [inline xla/service/cpu/onednn_pattern_utils.h:28]: Nit: The style guide prohibits anonymous namespaces in headers.

Comment (gaurides) [reply xla/service/cpu/onednn_pattern_utils.h:28]: Thanks for fixing the namespace

**Comment by gaurides** [comment]
@jreiffers if possible, can you also review this PR https://github.com/openxla/xla/pull/11832 ? it is similar.

**Pull request closed without merging**